axelor-sale/src/main/java/com/axelor/apps/sale/service/SaleOrderGeneratorService.java

```diff
@@ -23,10 +23,16 @@
 import com.axelor.apps.base.db.Currency;
 import com.axelor.apps.base.db.Partner;
 import com.axelor.apps.sale.db.SaleOrder;
+import com.axelor.apps.sale.rest.dto.SaleOrderLinePostRequest;
 import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.List;
 
 public interface SaleOrderGeneratorService {
   SaleOrder createSaleOrder(
       Partner clientPartner, Company company, Partner contactPartner, Currency currency)
       throws AxelorException, JsonProcessingException;
+
+  SaleOrder fetchAndAddSaleOrderLines(
```

Comment:
I think this method should not be in SaleOrderGeneratorService, but in a service for API related methods since we have a DTO in the parameters. For example a service called SaleOrderRestService.